### DISCUSSION TITLE
kmonad: Mismatch between length of 'defsrc' and deflayer <U_MEDIA>

### DISCUSSION THREAD

**Discussion by alloc33** [manna-harbour/miryoku#130] [Q&A] [answered]
I'm trying to force my MacBook use miryoku layout. MacBook Pro M1 pro. Monterey 12.5

Here is my built-in keyboard:
![IMG_2993 Small](https://user-images.githubusercontent.com/45129790/184071757-37714abf-6433-48b7-b046-6027705ac24e.jpeg)

I use GitHub actions with these values to build keymap for kmonad:
<img width="326" alt="Screen Shot 2022-08-11 at 1 41 29 PM" src="https://user-images.githubusercontent.com/45129790/184071902-bbb3daea-1ba3-4da5-adf4-e51df13a1b79.png">

And trying to run kmonad with path to .kbd file:
```
$ kmonad miryoku_kmonad.kbd
```

I got this:
<img width="917" alt="Screen Shot 2022-08-11 at 1 43 39 PM" src="https://user-images.githubusercontent.com/45129790/184073364-f5b4eaa3-a896-45ba-b6ef-c7940aad052e.png">

Tried to use `default` in mapping field but got the same kind of error. How can I fix that?

**Comment by manna-harbour** [comment, marked as answer]
Which version of KMonad are you using? Miryoku KMonad is written for the latest KMonad release, and also for the current master branch.

Also, to see if it's an unrecognised keycode, you could try replacing the media keycodes (`previoussong`, `vold`, `volu`, `nextsong`, `stopcd`, `playpause`, `mute`) with `XX` in your `miryoku_kmonad.kbd` in this section: https://github.com/manna-harbour/miryoku_kmonad/blob/c7b10aff11f7d7e4426fc0035af38d43e10d922d/src/build/miryoku_kmonad.kbd#L56-L59.

**Reply by alloc33** [reply]
Not possible to use kmonad master branch to install on M1 machine

**Reply by alloc33** [reply]
After disabling U_MEDIA layer completely I got this:
kmonad: Mismatch between length of 'defsrc' and deflayer <U_FUN>
Source length: 37
Layer length: 42

Probably 'lite' will not work, will try other option

**Reply by manna-harbour** [reply]
The `keycode-refactor` branch is not yet suitable for use, see https://github.com/kmonad/kmonad/pull/372. In particular, some keycodes have changed and aliases are missing, see https://github.com/kmonad/kmonad/issues/371. If you need to use that branch, you'd have to compare the keycodes between https://github.com/kmonad/kmonad/blob/master/src/KMonad/Keyboard/Keycode.hs and https://github.com/kmonad/kmonad/blob/keycode-refactor/src/KMonad/Util/Keyboard/Mac.hs to find the new names, then either fix the issue in KMonad by adding the missing aliases, or substitute the new names into either https://github.com/manna-harbour/miryoku_kmonad/blob/main/src/build/miryoku_kmonad.kbd, https://github.com/manna-harbour/miryoku_kmonad/blob/main/src/miryoku_babel/miryoku_layer_alternatives.h, or https://github.com/manna-harbour/miryoku_babel/blob/master/readme.org.
